Implement incognito mode

David-Development opened the pull request:
Fixes #773

Note the incognito icon on the bottom.

<img src="https://user-images.githubusercontent.com/4489723/65720647-b0644680-e0a8-11e9-9ab5-df326902ee52.png" width="300px" />

<img src="https://user-images.githubusercontent.com/4489723/65720658-b528fa80-e0a8-11e9-95f6-4d601cc08771.png" width="300px" />

David-Development commented:
As I didn't really liked the idea of having an overlay at the bottom, I tried using a bottom navigation bar instead. This also helps me use the app as I can't reach the read/unread and star/unstar button on my OnePlus anymore without moving the phone in my hand. The new design looks like this. Let me know what you guys think! @tobiasKaminsky @jancborchardt @stefan-niedermann 

<img width="200" src="https://user-images.githubusercontent.com/4489723/68026866-30d22480-fcb1-11e9-8543-4bd09c1a545d.png" />
<img width="200" src="https://user-images.githubusercontent.com/4489723/68026867-30d22480-fcb1-11e9-8ed0-3214aef903f9.png" />
<img width="200" src="https://user-images.githubusercontent.com/4489723/68026868-30d22480-fcb1-11e9-858d-fff38a437283.png" />

tobiasKaminsky commented:
I am quite unsure about it…
- the three buttons are maybe less used than the back button, so maybe switch it?
- what if you want to open the article? Then you still have to go all the way up…
- it is not common at all to have this at bottom

stefan-niedermann commented:
I also am very unsure... to be honest it just feels odd. The back-arrow is now nearly directly at the same position like the native android-back-triangle. It's a nice idea, and i know, mozilla is trying to do the same thing on firefox for android currently, but i would downvote this.

David-Development commented:
Thank you for your feedback guys!

> what if you want to open the article? Then you still have to go all the way up…

True, maybe we could add an floating action bar button that lets the user open the article

> it is not common at all to have this at bottom

isn't it? I saw some other readers going in the same direction.. and the bottom bar navigation is quite common now, isn't it? --> https://material.io/components/app-bars-bottom/#usage

> The back-arrow is now nearly directly at the same position like the native android-back-triangle.

Yes, valid point. It looks kind of odd on the emulator when I tested it.. 

But then again.. I feel like we need some changes here because screens are getting bigger and bigger and especially for the detail view, I'm using the buttons in the action-bar quite often and it's annoying to move the phone position around...

jancborchardt commented:
Hmm, so reading the original issue, it seems the solution proposed is actually quite good?

> Solution: Add a menu item – maybe a slider – which lets you choose whether to download external content at all. Options could be: Always / Only on WLAN / Never.

And that would just be in the settings. No indicator needed at all, nor a bottom bar.

----------------

(Regarding the bottom bar, I find it odd as well as said by others above already. :) Also I’m not sure what the square next to the star is … I’m guessing it’s a checkbox for having read it? – My point is if I am unsure then many people will have no clue. ;)

David-Development commented:
@jancborchardt Thank you for your feedback! :) I reverted the changes again and added a simple menu bottom in the Detail-View. That should be enough. Thanks again everyone for the feedback! 👍
